runtime/doc/lsp.txt

```diff
@@ -0,0 +1,265 @@
+                and false means the request failed.
+
+                The parameter {data} is a dictionary containing the possible
+                response data (or error) data as specified by the language
+                server protocol.
+
+                The callback will be executed from lua running:
+
+                    `vim.api.nvim_call_function('func_name', { success, data })`
+
+            [filetype]: The filetype associated with the server.
+            This will default to the current |filetype| if nothing is passed.
+
+            Configuration for the default server can be found at
+            |lsp-configuration|.
+
+                                                          *lsp#request_async()*
+lsp#request_async({method} [, {arguments} [, {callback} [, {filetype}]]])
+    Creates and sends a request to the server for [filetype]. This request
+    will not block and will call desired the [callback] upon completion.
+
+    Returns the request ID.
+
+    See |lsp#request()| for more information.
+
+================================================================================
+                                                         *lsp-default-callback*
+
+The following requests have built-in callbacks defined to handle the response
+in an idiomatic vim fashion.
+
+  textDocument/hover~
+  textDocument/references~
+  textDocument/definition~
+
+================================================================================
+                                                                 *lsp-autocmds*
+
+The language server implementation in Neovim will fire |autocmd-User| events
+upon each request and response so that users can attach to whenever these
+events are triggered.
+
+For the request 'textDocument/hover', the following autocmds will be fired:
+
+    Before sending the request:
+        'textDocument/hover/pre'
+
+    After sending the request:
+        'textDocument/hover/post'
+
+    Before handling the response:
+        'textDocument/hover/response'
+
+================================================================================
+                                                            *lsp-configuration*
+
+                                                             *lsp#server#add()*
+lsp#server#add({ftype}, {configuration}):
+    Used to add a language server.
+
+    This will add an autocmd that will start the server and handle any other
+    startup activities whenever the |FileType| event is triggered for {ftype}.
+
+    Parameters:~
+        {ftype}: A string or list of strings of filetypes to associate with
+        this server.
+
+        {configuration} (dictionary): The command to be sent to start the server
+            `name`        (string):   The name of the server
+            `command`     (string):   The command to start the server
+            `arguments`   (list):     Any arguments to pass to the server
```

Comment:
Why not specify the args conistenttly with jobstart? So `command` can either be a string, or a list of executable+arguments, and separate `arguments` is not needed. This would make the code simplers as this can be passed unmodified to `jobstart()`